src/utils/type-transform/Payload.ts

```diff
@@ -46,26 +52,55 @@ function deconstructAlpha41(str: string, throws = false): null | never {
 }
 
 const versions = new Set([deconstructAlpha40_Or_Alpha39, deconstructAlpha41])
-export function deconstructPayload(str: string): Payload | null
-export function deconstructPayload(str: string, throws: true): Payload
-export function deconstructPayload(str: string, throws = false): Payload | null {
-    for (const ver of versions) {
-        const result = ver(str, false)
-        if (throws === false) return result
-        if (result) return result
-        return ver(str, true)
+
+/**
+ * @type Decoder    defines decoder. if null, auto select decoder.
+ */
+type Decoder = SocialNetworkWorkerAndUI['payloadDecoder'] | null
+type Encoder = SocialNetworkWorkerAndUI['payloadEncoder']
+
+export function deconstructPayload(str: string, decoder: Decoder, opts?: { throws: false }): Payload | null
+export function deconstructPayload(str: string, decoder: Decoder, opts?: { throws: true }): Payload
+export function deconstructPayload(
+    str: string,
+    decoder: Decoder,
+    opts: { throws?: boolean } = { throws: false },
+): Payload | null {
+    defaults(opts, {
+        throws: false,
+    })
+
+    const decoders = (() => {
+        if (isNil(decoder)) {
+            const sto = []
+            for (const v of definedSocialNetworkWorkers) {
+                sto.push(v.payloadDecoder)
+            }
+            return sto
+        }
+        return [decoder]
+    })()
+
+    for (const versionDecoder of versions) {
+        for (const networkDecoder of decoders) {
+            if (isNil(networkDecoder(str))) continue
+            const result = versionDecoder(networkDecoder(str)!, false)
+            if (!isNil(result)) {
+                return result
+            }
+        }
     }
     if (str.includes('🎼') && str.includes(':||'))
-        if (throws) throw new TypeError(geti18nString('service_unknown_payload'))
+        if (opts.throws) throw new TypeError(geti18nString('service_unknown_payload'))
         else return null
-    if (throws) throw new TypeError(geti18nString('payload_not_found'))
+    if (opts.throws) throw new TypeError(geti18nString('payload_not_found'))
     else return null
 }
 
-export function constructAlpha40(data: Payload) {
-    return `🎼2/4|${data.ownersAESKeyEncrypted}|${data.iv}|${data.encryptedText}|${data.signature}:||`
+export function constructAlpha40(data: Payload, encoder: Encoder) {
+    return encoder(`🎼2/4|${data.ownersAESKeyEncrypted}|${data.iv}|${data.encryptedText}|${data.signature}:||`)
```

Comment:
Actually payload v40 can and only can be used in fb, so it's safe to ignore the encoder. (not requesting a change here)

Reply:
yes and it is an unused method.